app/models/proxy.rb

```diff
@@ -515,6 +519,19 @@ def service_mesh_integration?
     Service::DeploymentOption.service_mesh.include?(deployment_option)
   end
 
+  def find_or_create_proxy_config_affecting_change
+    proxy_config_affecting_change || create_proxy_config_affecting_change
+  end
+  alias affecting_change_history find_or_create_proxy_config_affecting_change
+  private :find_or_create_proxy_config_affecting_change
+
+  def pending_affecting_changes?
```

Comment:
Using this method (probably in the view) creates records right?
If yes, can we avoid that somehow?

Reply:
Well, if we have a proxy config, then we can also have an affecting change history I think. It also saves us from having to migrate data.

Reply:
fair enough